packages/aws-cdk-lib/aws-cloudwatch/README.md

```diff
@@ -387,6 +387,85 @@ only supports filtering by `unit` for Alarms, not in Dashboard graphs.
 Please see the following GitHub issue for a discussion on real unit
 calculations in CDK: https://github.com/aws/aws-cdk/issues/5595
 
+## Anomaly Detection Alarms
+
+CloudWatch anomaly detection applies machine learning algorithms to create a model of expected metric behavior. You can use anomaly detection to:
+
+- Detect anomalies with minimal configuration
+- Visualize expected metric behavior
+- Create alarms that trigger when metrics deviate from expected patterns
+
+### Creating an Anomaly Detection Alarm
+
+There are two ways to create an alarm based on anomaly detection:
+
+#### Method 1: Using the `AnomalyDetectionAlarm` class (Recommended)
+
+The most straightforward way is to use the dedicated `AnomalyDetectionAlarm` class, which is specifically designed for anomaly detection:
+
+```ts
+// Create a metric
+const metric = new cloudwatch.Metric({
+  namespace: 'AWS/EC2',
+  metricName: 'CPUUtilization',
+  statistic: 'Average',
+  period: Duration.minutes(5),
+});
+
+// Create an anomaly detection alarm
+const alarm = new cloudwatch.AnomalyDetectionAlarm(this, 'AnomalyAlarm', {
+  metric: metric,
+  evaluationPeriods: 3,
+  stdDevs: 2, // Number of standard deviations for the band (default: 2)
+  comparisonOperator: cloudwatch.ComparisonOperator.LESS_THAN_LOWER_OR_GREATER_THAN_UPPER_THRESHOLD,
+  alarmDescription: 'Alarm when metric is outside the expected band',
+});
+```
+
+This class handles all the complexity of setting up the anomaly detection band and configuring the alarm correctly.
+
+#### Method 2: Using the `createAnomalyDetectionAlarm` helper method
+
+You can also use the `createAnomalyDetectionAlarm` method on a `Metric` or `MathExpression` object:
+
+```ts
+// Create a metric
+const metric = new cloudwatch.Metric({
+  namespace: 'AWS/EC2',
+  metricName: 'CPUUtilization',
+  statistic: 'Average',
+  period: Duration.minutes(5),
+});
+
+// Create an anomaly detection alarm with default settings
+const alarm = metric.createAnomalyDetectionAlarm(this, 'AnomalyAlarm', {
```

Comment:
I still hate this and regret `metric.createAlarm()`. 

Please consider one more time to not add this? 😉